fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -179,9 +182,6 @@ def gatekeeper() -> None:
             # and execute the delayed actions (release gradients, unroll the buckets)
             Variable._execution_engine.queue_callback(optimizer._consume_work_handles)
 
-            # Reset all the grad reduce and bucket state flags
```

Comment:
this was risky in that maybe that we never go there, if all grads are not materialized. Moved to the beginning of the FW, which will always happen